text: Change log entry for 0.3.0 (#153)
BlockstreamResearch/bip-frost-dkg: README.md

License: CC0-1.0
License-Code: MIT
Discussion: 2024-07-08: https://groups.google.com/g/bitcoindev/c/HE3HSnGTpoQ/m/euZvPxKeAQAJ

> Version: 0.4.0

Reviewer: ```suggestion
Version: 0.3.0-dev
```

Sorry, I have no idea why I suggested 0.4.0 instead of 0.3.0. :D

Author: Oh, my apologies too! Actually, I should have asked why it isn't 0.3.0...

The `MINOR` version is incremented whenever the inputs or the output of an algorithm changes in a backward-compatible way or new backward-compatible functionality is added.
The `PATCH` version is incremented for other noteworthy changes (bug fixes, test vectors, important clarifications, etc.).

* *0.4.0* (2026-07-29): Major expansion of the reference implementation with message serialization, comprehensive test coverage, and API refinements:
  * Add message serialization for all protocol messages, enabling byte-level interoperability between implementations.
  * Add optional recovery acknowledgment round: `participant_recovery_ack_sign` and `participant_recovery_acks_verify` let participants confirm receipt of recovery data before use.
  * Add comprehensive test vectors for all public API functions, including per-participant edge cases, boundary configurations, and exhaustive error scenarios.
  * Allow single-participant sessions (`n = 1`).

>   * Fix blame guarantee for public nonces: pubnonces are now validated as curve points, and failures are attributed to the sending participant instead of surfacing as an unattributed `ValueError`.

Reviewer: ```suggestion
  * Fix blaming for invalid public nonces: pubnonces are now validated as curve points, and failures are attributed to the sending participant instead of surfacing as an unattributed `ValueError`.
```
Just style. I don't think we fix the guarantee. The guarantee was already there and intentional, but the code didn't hold up to it.

>   * Fix blame in `participant_finalize`: an invalid certificate signature is now attributed to the coordinator, since the coordinator verifies all signatures before broadcasting the certificate.

Reviewer: ```suggestion
  * Fix blaming in `participant_finalize`: an invalid certificate signature is now attributed to the coordinator, since the coordinator verifies all signatures before broadcasting the certificate.
```

  * Fix TapTweak to use BIP 341's x-only serialization.
  * Rewrite Setup section to make the authentic-hostpubkeys requirement explicit as a precondition of the session.

>   * Fix EncPedPop description to match the reference implementation, fix typos, and update the preamble per BIP 3.

Reviewer: "update the preamble per BIP 3" should probably be a separate item even though it happened in the same PR.

(And I think we can omit "fix types").

  * Expose randomness as explicit parameters: `participant_step1` takes `random` and `participant_step2` takes `aux_rand`, which were previously generated internally.
  * Remove message classes from the public API; protocol messages are passed as `bytes`.
  * Remove `ParticipantMsgParseError` and `CoordinatorMsgParseError` from the public API.
  * Add `RandomnessError`, `InvalidRecoveryAckError`, and `FaultyParticipantError` to the public API.

>   * Rename `secp256k1proto` to `secp256k1lab` and separate it as an independent subtree for reuse across projects.

Reviewer: Could here mention that the upstream repo is at https://github.com/secp256k1lab/secp256k1lab .

The `MINOR` version is incremented whenever the inputs or the output of an algorithm changes in a backward-compatible way or new backward-compatible functionality is added.
The `PATCH` version is incremented for other noteworthy changes (bug fixes, test vectors, important clarifications, etc.).

> * *0.4.0* (2026-07-29): Major expansion of the reference implementation with message serialization, comprehensive test coverage, and API refinements:

Reviewer: still 0.4.0 here :) 

(and maybe set the date to tomorrow, which is hopefully the day on which we open the PR)

Author: Ah, missed that... Fixed now, thanks!
